### ISSUE TITLE
V2000 chiral flag does not seem to be read by Chem.MolFromMolBlock()

### ISSUE THREAD

**Issue by GLPG-GT** [rdkit/rdkit#6062]
**Describe the bug**
When converting a V2000 molfile to molecule by Chem.MolFromMolBlock(), the 'chiral flag', i.e. the 5th element of the [counts line block](https://en.wikipedia.org/wiki/Chemical_table_file#Counts_line_block_specification), which switches the interpretation between 1 = absolute and 0 = relative = racemic, is not taken into account, which means both the consequent stereo groups detection and conversion to CXSMILES is incorrect.
By using a V3000 molfile, the issue seems to be absent.

**To Reproduce**
```
from rdkit import Chem
from rdkit.Chem.EnumerateStereoisomers import EnumerateStereoisomers, StereoEnumerationOptions

# Absolute molecule
m = Chem.MolFromMolBlock('''
  ACCLDraw02072309142D

 16 17  0  0  1  0  0  0  0  0999 V2000
    5.1080   -4.6152    0.0000 S   0  0  0  0  0  0  0  0  0  0  0  0
    5.5937   -4.2632    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.4089   -3.6918    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.7621   -3.2076    0.0000 C   0  0  2  0  0  0  0  0  0  0  0  0
    6.3585   -3.2710    0.0000 C   0  0  1  0  0  0  0  0  0  0  0  0
    6.7116   -2.7861    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    6.6019   -3.8192    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    5.5187   -2.6594    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
    4.8091   -3.6918    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.6232   -4.2621    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
    4.4571   -3.2062    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.8605   -3.2683    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.5085   -2.7825    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.7531   -2.2350    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.3497   -2.1728    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.7017   -2.6585    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0  0  0  0
  2  3  2  0  0  0  0
  4  3  1  0  0  0  0
  5  4  1  0  0  0  0
  6  5  1  0  0  0  0
  5  7  1  1  0  0  0
  4  8  1  1  0  0  0
  9  3  1  0  0  0  0
  9 10  2  0  0  0  0
 10  1  1  0  0  0  0
  9 11  1  0  0  0  0
 12 11  2  0  0  0  0
 13 12  1  0  0  0  0
 14 13  2  0  0  0  0
 15 14  1  0  0  0  0
 16 15  2  0  0  0  0
 11 16  1  0  0  0  0
M  END
''')

Chem.MolToCXSmiles(m)
#'C[C@@H](O)[C@H](Cl)c1csnc1-c1ccccc1 |(6.7116,-2.7861,;6.3585,-3.271,;6.6019,-3.8192,;5.7621,-3.2076,;5.5187,-2.6594,;5.4089,-3.6918,;5.5937,-4.2632,;5.108,-4.6152,;4.6232,-4.2621,;4.8091,-3.6918,;4.4571,-3.2062,;3.8605,-3.2683,;3.5085,-2.7825,;3.7531,-2.235,;4.3497,-2.1728,;4.7017,-2.6585,)|'

# racemic molecule in V2000 format
m_rac = Chem.MolFromMolBlock('''
  ACCLDraw02072309202D

 16 17  0  0  0  0  0  0  0  0999 V2000
    4.6923   -7.0446    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.4477   -7.5923    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.8511   -7.6545    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.4991   -7.1687    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    3.7437   -6.6211    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.3403   -6.5590    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.7997   -8.0780    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    4.6138   -8.6482    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
    5.0986   -9.0013    0.0000 S   0  0  0  0  0  0  0  0  0  0  0  0
    5.5843   -8.6493    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.3996   -8.0780    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
    5.7527   -7.5938    0.0000 C   0  0  1  0  0  0  0  0  0  0  0  0
    5.5093   -7.0455    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
    6.3491   -7.6571    0.0000 C   0  0  2  0  0  0  0  0  0  0  0  0
    6.5925   -8.2054    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
    6.7022   -7.1723    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  2  1  1  0  0  0  0
  3  2  2  0  0  0  0
  4  3  1  0  0  0  0
  5  4  2  0  0  0  0
  6  5  1  0  0  0  0
  1  6  2  0  0  0  0
  7  2  1  0  0  0  0
  7  8  2  0  0  0  0
  8  9  1  0  0  0  0
  9 10  1  0  0  0  0
 10 11  2  0  0  0  0
  7 11  1  0  0  0  0
 12 11  1  0  0  0  0
 12 13  1  1  0  0  0
 14 12  1  0  0  0  0
 14 15  1  1  0  0  0
 16 14  1  0  0  0  0
M  END
''')

Chem.MolToCXSmiles(m_rac)
#'C[C@@H](O)[C@H](Cl)c1csnc1-c1ccccc1 |(6.7022,-7.1723,;6.3491,-7.6571,;6.5925,-8.2054,;5.7527,-7.5938,;5.5093,-7.0455,;5.3996,-8.078,;5.5843,-8.6493,;5.0986,-9.0013,;4.6138,-8.6482,;4.7997,-8.078,;4.4477,-7.5923,;3.8511,-7.6545,;3.4991,-7.1687,;3.7437,-6.6211,;4.3403,-6.559,;4.6923,-7.0446,)|'

m_rac.GetStereoGroups()[0].GetGroupType()
#---------------------------------------------------------------------------
#IndexError                                Traceback (most recent call last)
#Cell In [36], line 1
#----> 1 m_rac.GetStereoGroups()[0].GetGroupType()
#
#IndexError: Index out of range

# racemic molecule in V3000 format
m_rac_V3000 = Chem.MolFromMolBlock('''
  ACCLDraw02072309272D

  0  0  0     0  0            999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 16 17 0 0 0
M  V30 BEGIN ATOM
M  V30 1 C 4.6923 -7.0446 0 0 
M  V30 2 C 4.4477 -7.5923 0 0 
M  V30 3 C 3.8511 -7.6545 0 0 
M  V30 4 C 3.4991 -7.1687 0 0 
M  V30 5 C 3.7437 -6.6211 0 0 
M  V30 6 C 4.3403 -6.559 0 0 
M  V30 7 C 4.7997 -8.078 0 0 
M  V30 8 N 4.6138 -8.6482 0 0 
M  V30 9 S 5.0986 -9.0013 0 0 
M  V30 10 C 5.5843 -8.6493 0 0 
M  V30 11 C 5.3996 -8.078 0 0 
M  V30 12 C 5.7527 -7.5938 0 0 CFG=1 
M  V30 13 Cl 5.5093 -7.0455 0 0 
M  V30 14 C 6.3491 -7.6571 0 0 CFG=2 
M  V30 15 O 6.5925 -8.2054 0 0 
M  V30 16 C 6.7022 -7.1723 0 0 
M  V30 END ATOM
M  V30 BEGIN BOND
M  V30 1 1 2 1 
M  V30 2 2 3 2 
M  V30 3 1 4 3 
M  V30 4 2 5 4 
M  V30 5 1 6 5 
M  V30 6 2 1 6 
M  V30 7 1 7 2 
M  V30 8 2 7 8 
M  V30 9 1 8 9 
M  V30 10 1 9 10 
M  V30 11 2 10 11 
M  V30 12 1 7 11 
M  V30 13 1 12 11 
M  V30 14 1 12 13 CFG=1 
M  V30 15 1 14 12 
M  V30 16 1 14 15 CFG=1 
M  V30 17 1 16 14 
M  V30 END BOND
M  V30 BEGIN COLLECTION
M  V30 MDLV30/STERAC2 ATOMS=(2 12 14)
M  V30 END COLLECTION
M  V30 END CTAB
M  END
''')

Chem.MolToCXSmiles(m_rac_V3000)
#'C[C@@H](O)[C@H](Cl)c1csnc1-c1ccccc1 |(6.7022,-7.1723,;6.3491,-7.6571,;6.5925,-8.2054,;5.7527,-7.5938,;5.5093,-7.0455,;5.3996,-8.078,;5.5843,-8.6493,;5.0986,-9.0013,;4.6138,-8.6482,;4.7997,-8.078,;4.4477,-7.5923,;3.8511,-7.6545,;3.4991,-7.1687,;3.7437,-6.6211,;4.3403,-6.559,;4.6923,-7.0446,),wU:1.1,3.3,&1:1,3|'

m_rac_V3000.GetStereoGroups()[0].GetGroupType()
#rdkit.Chem.rdchem.StereoGroupType.STEREO_AND
```

**Expected behavior**
The chiral flag in V2000 molfile text should be correctly converted to 'r' enhanced stereo, or equivalently, a single '&' stereo group should be applied to all sp3 stereocenters.

**Screenshots**
Not applicable.

**Configuration (please complete the following information):**
 - RDKit version: 2022.09.1
 - OS: Ubuntu
 - Python version (if relevant): 3.9.15
 - Are you using conda? yes
 - If you are using conda, which channel did you install the rdkit from? NA
 - If you are not using conda: how did you install the RDKit? NA

**Additional context**
Not applicable.

**Comment by greglandrum** [comment]
@GLPG-GT : the current behavior, ignoring the chiral flag, is by design. There are multiple different interpretations of what that flag actually means out there, so it's impossible to do anything sensible with it that won't be wrong a large fraction of the time. So we ignore it.

The state of the chiral flag is, however, actually read in and is avaialble as a molecular propety:
```
In [11]: m = Chem.MolFromMolBlock('''foo
    ...:   Mrv2219 02072317202D
    ...:
    ...:   5  4  0  0  1  0            999 V2000
    ...:    -2.3884    1.1384    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0  
    ...:    -1.6739    1.5509    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0  
    ...:    -3.1029    1.5509    0.0000 Br  0  0  0  0  0  0  0  0  0  0  0  0  
    ...:    -2.3884    0.3134    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0  
    ...:    -3.1029    0.7259    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0  
    ...:   1  3  1  0  0  0  0
    ...:   1  4  1  0  0  0  0
    ...:   1  5  1  0  0  0  0
    ...:   1  2  1  1  0  0  0
    ...: M  END
    ...: ''')

In [12]: m.GetProp('_MolFileChiralFlag')
Out[12]: '1'
```

If you have an interpretation of the chiral flag which makes sense for the files that you process, then you can check that property value and make the appropriate modifications to your molecule.

**Comment by GLPG-GT** [comment]
Thank you @greglandrum !
And sorry for tagging this as a bug.
I suspected that it wasn't, and you are right, of course, the V2000 chiral flag is not always intentionally set by the originators of CTAB's.
This report was related to my other question https://github.com/rdkit/rdkit/discussions/6060 .
I am trying to enumerate all the stereoisomers that are implied by a given CTAB.
EnumerateStereoisomers works 100% fine when I start from V3000 CTAB's.
But I have mostly V2000 CTAB's in our DB, which we curated to ensure that the chiral flag does mean what it should.
So I was simply looking for a way to make it work with molecules originating from Chem.MolFromMolBlock() applied to V2000, too.
